### ISSUE TITLE
Please add support nullable types (PHP 7.1) in ProxyGenerator

### ISSUE THREAD

**Issue by l-vitaly** [doctrine/common#731]
Example: 

```
class Entity {
  public function getName() : ?string
  {
    return $this->name;
  }

  public function setName(?string $name)
  {
    $this->name = $name;
  }
}
```

**Comment by Ocramius** [comment]
We'll also need `void`, which is a mess

**Comment by Majkl578** [comment]
This is fixed by #734.

**Comment by Ocramius** [comment]
@Majkl578 awesome!
